sycl/test-e2e/Annotated_usm/annotated_usm.cpp

```diff
@@ -0,0 +1,160 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+// E2E test for annotated USM allocation
+
+#include "sycl/sycl.hpp"
+#include <iostream>
+
+// clang-format on
+
+#define CHECK_USM_KIND(ap, Kind)                                               \
+  assert(ap.get() != nullptr && sycl::get_pointer_type(ap.get(), Ctx) == Kind)
+#define CHECK_ALIGN(ap, N)                                                     \
+  assert(ap.get() != nullptr && ((uintptr_t)ap.get() % N) == 0)
+
+using namespace sycl::ext::oneapi::experimental;
+using namespace sycl::ext::intel::experimental;
+using alloc = sycl::usm::alloc;
+
+constexpr int N = 10;
+
+void TestUsmKind(sycl::queue &q) {
+  const sycl::context &Ctx = q.get_context();
+  auto dev = q.get_device();
+
+  // Allocation funtions with the USM kind in the function name
+  properties P1{conduit, stable};
+  auto APtr1 = malloc_device_annotated<int>(N, q, P1);
+  CHECK_USM_KIND(APtr1, alloc::device);
+
+  auto APtr2 = aligned_alloc_device_annotated(128, N, q);
+  CHECK_USM_KIND(APtr2, alloc::device);
+
+  auto APtr3 = malloc_host_annotated<int>(N, q);
+  CHECK_USM_KIND(APtr3, alloc::host);
+
+  if (dev.has(sycl::aspect::usm_shared_allocations)) {
+    auto APtr4 = malloc_shared_annotated(N, q);
+    CHECK_USM_KIND(APtr4, alloc::shared);
+    free(APtr4, q);
+  }
+
+  // Parameterized functions: the USM kind is specified by an argument
+  properties P2{conduit, cache_config{large_slm}};
+  auto APtr5 = malloc_annotated<int>(N, q, alloc::device, P2);
```

Comment:
Should we check that APtr5 has only compile time constant properties and that cache_config was not transferred to it? (That is the intent right?)

Reply:
this is checked in lit test